### PR TITLE
Get role into 2023

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY

- Updated the role to best practices
- Added a very important feature. (cowsay)
- Added a very, very basic molecule configuration.
- Removed support for centos/rhel based systems .
- Cleaned up garbage
- Updated meta information

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

### CODE REVIEW

File: tasks/installation.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
 - name: install motd related packages
-  package:
+  ansible.builtin.package:
     name: '{{ motd_packages }}'
     state: present
+
+- name: install cowsay packages
+  ansible.builtin.package:
+    name: 'cowsay'
+    state: present
```
Comment (hairmare) [inline tasks/installation.yml]: i'm here for cowsay, but what if i already have a non package based installation of cowsay that supports offensive mode?

Maybe installing cowsay needs to be skippable?
```suggestion
- name: install cowsay packages
  ansible.builtin.package:
    name: 'cowsay'
    state: present
  when: motd_cowsay is true
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## [Requirements](#requirements)
+
+- None
+
+## [Compatibility](#compatibility)
+
+This role is only compatible with debian bullseye and bookworm.
+
+The minimum version of Ansible required is 2.12.
```
Comment (s3lph) [inline README.md:50]: Can you add a footer (see README.rst) in the README.md (of course `s/Adfinis SyGroup/Adfinis/`)

File: tasks/configuration.yml [hunk 3]
```diff
@@ -1,7 +1,21 @@
 ---
 
+- name: wrap /etc/issue message in cowsay
+  ansible.builtin.shell:
+    cmd: "/usr/games/cowsay"
+    stdin: "{{ motd_issue_message }}"
```
Comment (s3lph) [inline tasks/configuration.yml:6]:
```suggestion
    stdin: "{{ motd_issue_message }}"
  changed_when: false
```

File: tasks/configuration.yml [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
+  register: motd_issue_cowsay_message
+  when: motd_cowsay is true
+
+- name: wrap /etc/motd message in cowsay
+  ansible.builtin.shell:
+    cmd: "/usr/games/cowsay"
+    stdin: "{{ motd_message }}"
```
Comment (s3lph) [inline tasks/configuration.yml:14]:
```suggestion
    stdin: "{{ motd_message }}"
  changed_when: false
```

File: tasks/configuration.yml [hunk 5]
```diff
@@ -1,7 +1,21 @@
 ---
 
+- name: wrap /etc/issue message in cowsay
+  ansible.builtin.shell:
```
Comment (s3lph) [inline tasks/configuration.yml]: Any reason this is `shell` and not `command`?

**Review by s3lph — CHANGES_REQUESTED** [review]

File: tasks/configuration.yml (hunk 5 above)
Comment (ghost) [reply tasks/configuration.yml]: Resolved in [bddc8c0](https://github.com/adfinis/ansible-role-motd/pull/5/commits/bddc8c0e0077830e0e60bb08b6f5e80442b4f870)

File: tasks/configuration.yml (hunk 3 above)
Comment (ghost) [reply tasks/configuration.yml:6]: Resolved in [47832d3](https://github.com/adfinis/ansible-role-motd/pull/5/commits/47832d3b079677522019d743d7c61d6c355ebf28)

File: tasks/configuration.yml (hunk 4 above)
Comment (ghost) [reply tasks/configuration.yml:14]: Resolved in [47832d3](https://github.com/adfinis/ansible-role-motd/pull/5/commits/47832d3b079677522019d743d7c61d6c355ebf28)

File: tasks/installation.yml (hunk 1 above)
Comment (ghost) [reply tasks/installation.yml]: Resolved in [401f973](https://github.com/adfinis/ansible-role-motd/pull/5/commits/401f97303d93b763483a0ace109d64833015fa95)

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md:50]: Resolved in [44252b8](https://github.com/adfinis/ansible-role-motd/pull/5/commits/44252b85333b2fba23e77d5bc9c83b0644d78a3a)

**Review by s3lph — CHANGES_REQUESTED** [review]
LGTM in general.

Though I'm wondering why you are removing support for RedHat based systems. Literally the only difference in this role between Debian and RedHat is the name of the package providing the motd files (`base-files` vs `setup`), which this role would not even need to install, since it's already present on all installations. Any other differences (e.g. SELinux contexts for RHEL) are still part of the role anyway.

I'd propose to:
- Remove motd_packages and its installation entirely (unless I'm missing something and there is a legitimate reason to manually install the package)
- Move `motd_file`, `motd_issue_file` and `motd_issue_net_file` from vars to defaults.
- Re-add EL to galaxy.yml metadata.

**Comment by ghost** [comment]
> LGTM in general.
> 
> Though I'm wondering why you are removing support for RedHat based systems. Literally the only difference in this role between Debian and RedHat is the name of the package providing the motd files (`base-files` vs `setup`), which this role would not even need to install, since it's already present on all installations. Any other differences (e.g. SELinux contexts for RHEL) are still part of the role anyway.
> 
> I'd propose to:
> 
>     * Remove motd_packages and its installation entirely (unless I'm missing something and there is a legitimate reason to manually install the package)
> 
>     * Move `motd_file`, `motd_issue_file` and `motd_issue_net_file` from vars to defaults.
> 
>     * Re-add EL to galaxy.yml metadata.

Thanks for your input!

Ive removed EL support in the first place as a mistake and re-added it now.
Based on your input, ive also moved the variables to default and removed the tasks to install the motd packages.

File: meta/main.yml [hunk 6]
```diff
@@ -15,5 +15,9 @@ galaxy_info:
       versions:
         - bullseye
         - bookworm
+    - name: EL
+      versions:
+        - "7"
+        - "8"
```
Comment (s3lph) [inline meta/main.yml:21]:
```suggestion
        - "8"
        - "9"
```

**Review by s3lph — CHANGES_REQUESTED** [review]
Add EL9, otherwise I think it's good to go.

**Review by s3lph — APPROVED** [review]

**Pull request merged**